### PR TITLE
feat(plugin): Pinecone

### PR DESCRIPTION
**Pull request by BominRahmani**
Waiting on bug fix from Pinecone before fully setting up the PR for this. Currently going to stay as Draft.

##### Checklist
- [ ] Changes are tested
- [ ] CI has passed

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -21,6 +21,7 @@ opentelemetry-java-contrib-jmx-metrics.jar
 VERSION.txt
 release_deps
 /tmp
+tmp
```
Comment (cpheps) [inline .gitignore:24]: We already ignore temp can you condense this line and the one above with `**/tmp` I think that should ignore either path.

File: plugins/pinecone.yaml [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+template: |
+  receivers:
+    prometheus/pinecone:
+      config:
+        scrape_configs:
+          - job_name: 'pinecone'
+            scrape_interval: '{{ .scrape_interval }}'
+            authorization:
+              credentials: '{{ .api_key }}'
+            scheme: https
```
Comment (cpheps) [inline plugins/pinecone.yaml:26]: we're forcing TLS here will it always work or is there a locally hosted version of pinecone where they don't require TLS?

File: plugins/pinecone.yaml [hunk 3]
```diff
@@ -0,0 +1,103 @@
+version: 0.0.1
+title: Pinecone Metrics
+description: Metrics receiver for Pinecone
+parameters:
+  - name: environment
+    description: Environment of your project  (example us-east4-gcp)
```
Comment (cpheps) [inline plugins/pinecone.yaml:6]: Is this a gcp zone or is it broader than that? Is this easy for users to find?

**Review by cpheps — CHANGES_REQUESTED** [review]
Make sure to run `make create-plugin-docs`

File: plugins/pinecone.yaml [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+      - context: metric
+        statements:
+        - set(description, "Number of items per pod in the index.") where name == "pinecone.vector.count"
+        - set(unit, "{items}") where name == "pinecone.vector.count"
+
+        - set(description, "Number of data plane calls made by clients.") where name == "pinecone.request.count"
+        - set(unit, "{requests}") where name == "pinecone.request.count"
+
+        - set(description, "Number of data plane calls made by clients that resulted in errors.") where name == "pinecone.request.error.count"
+        - set(unit, "requests") where name == "pinecone.request.error.count"
```
Comment (BinaryFissionGames) [inline plugins/pinecone.yaml:91]:
```suggestion
        - set(unit, "{requests}") where name == "pinecone.request.error.count"
```

File: plugins/pinecone.yaml [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+        - set(description, "Distribution of server-side processing latency for pinecone data plane calls.") where name == "pinecone.request.latency"
+        - set(unit, 1) where name == "pinecone.request.latency"
```
Comment (BinaryFissionGames) [inline plugins/pinecone.yaml:94]: Latency is in seconds here, it looks like?
```suggestion
        - set(unit, "s") where name == "pinecone.request.latency"
```

**Review by BinaryFissionGames — COMMENTED** [review]
Could we change the filename from `pinecone.yaml` to `pinecone_metrics.yaml`?

File: plugins/pinecone.yaml (hunk 2 above)
Comment (jsirianni) [reply plugins/pinecone.yaml:26]: I could be wrong, but I believe Pinecone is a saas, which would always have TLS.

**Review by jsirianni — COMMENTED** [review]
Once this is working, I would be curious to see the metric Resource Attributes. We want to be sure they are unique. For example, the user configures two pinecone plugin receivers to collect metrics from two different accounts. Will the resulting metrics be duplicates or will their resource attributes guarantee uniqueness?

**Comment by cpheps** [comment]
Closing this due to staleness we can reopen if we need it.

**Pull request closed without merging**